File: api/src/main/java/fr/kksdev/budget/api/dto/response/UserPreferenceResponse.java
package fr.kksdev.budget.api.dto.response;

import fr.kksdev.budget.api.enums.Currency;
import fr.kksdev.budget.api.enums.Feature;
import fr.kksdev.budget.api.enums.NotificationType;
import fr.kksdev.budget.api.enums.TextScale;

import java.util.List;
import java.util.UUID;

public record UserPreferenceResponse(
        List<Feature> enabledFeatures,
        List<Feature> navOrder,
        UUID shopAccountId,
        Boolean includeShopInBalance,
        List<Currency> currencies,
        List<NotificationType> enabledNotificationTypes,
        String timezone,
        TextScale textScale
) {}

File: flutter/lib/src/data/remote/dtos/user_preference_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:k_budget/src/domain/enums/enums.dart';

part 'user_preference_response.freezed.dart';
part 'user_preference_response.g.dart';

@freezed
class UserPreferenceResponse with _$UserPreferenceResponse {
  const factory UserPreferenceResponse({
    required List<Feature> enabledFeatures,
    required List<Feature> navOrder,
    @Default(['EUR']) List<String> currencies,
    @Default([]) List<NotificationType> enabledNotificationTypes,
    @Default('Europe/Paris') String timezone,
    String? textScale,
  }) = _UserPreferenceResponse;

  factory UserPreferenceResponse.fromJson(Map<String, dynamic> json) =>
      _$UserPreferenceResponseFromJson(json);
}
